src/App.tsx

```diff
@@ -221,24 +230,68 @@ async function doChat(
       },
     );
 
+    let content = '';
+    let thinkingEnd = -1;
+
     for await (const chunk of stream) {
       if (progressStore.getSnapshot() !== '') {
         progressStore.setText('');
       }
 
       if (isContentChunk(chunk)) {
-        messageStore.appendText(chunk.choices[0].delta.content as string);
+        content += chunk['choices'][0]['delta']['content'];
+
+        switch (modelConfig.id) {
+          case 'deepseek-r1': {
+            const openTag = '<think>';
+            const closeTag = '</think>';
+
+            if (thinkingEnd === -1) {
+              thinkingEnd = content.indexOf(closeTag);
+              // stream the thoughts part to the thinking store
+              if (content.length >= openTag.length) {
+                thinkingStore.setText(
+                  content.slice(
+                    openTag.length,
+                    thinkingEnd === -1 ? content.length : thinkingEnd,
+                  ),
+                );
+              }
+            }
+
+            if (thinkingEnd !== -1) {
+              // stream the response part to the message store
+              messageStore.setText(
+                content.slice(thinkingEnd + closeTag.length),
+              );
+            }
+            break;
+          }
+
+          default: {
+            messageStore.setText(content);
+            break;
+          }
+        }
```

Comment:
Can you add a comment that it'd be nice to refactor this to a helper similar to `assembleToolCallsFromStream` below? It'll help keep the App lean and it will be easier to test.